lib/matplotlib/axis.py

```diff
@@ -1114,14 +1114,15 @@ def get_tightbbox(self, renderer, *, ignore_label=None):
         if self.label.get_visible():
             bb = self.label.get_window_extent(renderer)
             # for constrained/tight_layout, we want to ignore the label's
-            # width because the adjustments they make can't be improved.
+            # width/height because the adjustments they make can't be improved.
             # this code collapses the relevant direction
-            if ignore_label == 'x' and bb.width > 0:
-                bb.x0 = (bb.x0 + bb.x1) / 2 - 0.5
-                bb.x1 = bb.x0 + 1.0
-            elif ignore_label == 'y' and bb.height > 0:
-                bb.y0 = (bb.y0 + bb.y1) / 2 - 0.5
-                bb.y1 = bb.y0 + 1.0
+            if for_layout_only:
+                if self.axis_name == "x" and bb.width > 0:
```

Comment:
... and I guess I still don't particularly like the introspection.  But, again if other folks think the consistency is more important than explicitness I won't block.

Reply:
Maybe a bit of paranoia here with `axis_name = getattr(self, 'axis_name', '')`?